File: pset1/cash/cash.c
#include<cs50.h>
#include<stdio.h>
#include<math.h>


int main(void){
    
    // ask the user for the changed required
    float x;
    do{
        x=get_float("Change owed: ");
        
    }while(x<=0);
    
    //greedy algo to find minimum number of coins 
    int cents = round(x*100);
    int count =0;
    while(cents>=25){
        cents -= 25;
        count++;
        
    }
     while(cents>=10){
        cents -= 10;
        count++;
    }
     while(cents>=5){
        cents -= 5;
        count++;
    }
     while(cents>=1){
        cents -= 1;
        count++;
    }
    
    // print the minimum number of coins
    printf("%i\n", count);
    
    
}

File: pset6/cash.py
from cs50 import get_string, get_int, get_float

def main():
    while True:
        x = get_float("Change owed: ")
        if x>=0:
            break
    cents = round(x*100)
    count = 0
    while cents >=25 :
        cents -= 25
        count += 1
    while cents >=10 :
        cents -= 10
        count += 1
    while cents >=5 :
        cents -= 5
        count += 1
    while cents >=1 :
        cents -= 1
        count += 1
    print(count)

if __name__ == "__main__":
    main()
